streaming/base/dataset.py

```diff
@@ -1208,7 +1222,7 @@ def _each_sample_id(self, it: _Iterator) -> Iterator[int]:
 
             # Background thread crashed, terminate the main process
             if self._event.is_set():
-                raise RuntimeError('Background thread failed. Check other traceback.')
```

Comment:
Why don't we raise exception anymore for `if self._event.is_set():` and the other ones you added?

Reply:
The event is only set by the thread and if thread fails, we are propagating the error back to the main process to display as stderr. In the main process, we just need to check if the event has been set or not and come out of the loop, if we also raise an exception in the main process then there is a stack of traceback which confuses the user.